op-chain-ops/genesis/config.go

```diff
@@ -119,34 +93,50 @@ type DeployConfig struct {
-	L2GenesisDeltaTimeOffset *hexutil.Uint64 `json:"l2GenesisDeltaTimeOffset,omitempty"`
-	// L2GenesisEcotoneTimeOffset is the number of seconds after genesis block that Ecotone hard fork activates.
-	// Set it to 0 to activate at genesis. Nil to disable Ecotone.
-	L2GenesisEcotoneTimeOffset *hexutil.Uint64 `json:"l2GenesisEcotoneTimeOffset,omitempty"`
-	// L2GenesisFjordTimeOffset is the number of seconds after genesis block that Fjord hard fork activates.
-	// Set it to 0 to activate at genesis. Nil to disable Fjord.
-	L2GenesisFjordTimeOffset *hexutil.Uint64 `json:"l2GenesisFjordTimeOffset,omitempty"`
-	// L2GenesisInteropTimeOffset is the number of seconds after genesis block that the Interop hard fork activates.
-	// Set it to 0 to activate at genesis. Nil to disable Interop.
-	L2GenesisInteropTimeOffset *hexutil.Uint64 `json:"l2GenesisInteropTimeOffset,omitempty"`
 	// L2GenesisBlockExtraData is configurable extradata. Will default to []byte("BEDROCK") if left unspecified.
 	L2GenesisBlockExtraData []byte `json:"l2GenesisBlockExtraData"`
+	// Note that there is no L2 genesis timestamp:
+	// This is instead configured based on the timestamp of "l1StartingBlockTag".
+}
+
+var _ ConfigChecker = (*L2GenesisBlockDeployConfig)(nil)
+
+func (d *L2GenesisBlockDeployConfig) Check(log log.Logger) error {
+	if d.L2GenesisBlockGasLimit == 0 {
+		return fmt.Errorf("%w: L2 genesis block gas limit cannot be 0", ErrInvalidDeployConfig)
+	}
+	// When the initial resource config is made to be configurable by the DeployConfig, ensure
+	// that this check is updated to use the values from the DeployConfig instead of the defaults.
+	if uint64(d.L2GenesisBlockGasLimit) < uint64(MaxResourceLimit+SystemTxMaxGas) {
+		return fmt.Errorf("%w: L2 genesis block gas limit is too small", ErrInvalidDeployConfig)
+	}
+	if d.L2GenesisBlockBaseFeePerGas == nil {
+		return fmt.Errorf("%w: L2 genesis block base fee per gas cannot be nil", ErrInvalidDeployConfig)
+	}
+	return nil
+}
+
+type OwnershipDeployConfig struct {
 	// ProxyAdminOwner represents the owner of the ProxyAdmin predeploy on L2.
 	ProxyAdminOwner common.Address `json:"proxyAdminOwner"`
 	// FinalSystemOwner is the owner of the system on L1. Any L1 contract that is ownable has
 	// this account set as its owner.
 	FinalSystemOwner common.Address `json:"finalSystemOwner"`
-	// SuperchainConfigGuardian represents the GUARDIAN account in the SuperchainConfig. Has the ability to pause withdrawals.
-	SuperchainConfigGuardian common.Address `json:"superchainConfigGuardian"`
+}
+
+var _ ConfigChecker = (*OwnershipDeployConfig)(nil)
+
+func (d *OwnershipDeployConfig) Check(log log.Logger) error {
+	if d.FinalSystemOwner == (common.Address{}) {
+		return fmt.Errorf("%w: FinalSystemOwner cannot be address(0)", ErrInvalidDeployConfig)
+	}
+	if d.ProxyAdminOwner == (common.Address{}) {
+		return fmt.Errorf("%w: ProxyAdminOwner cannot be address(0)", ErrInvalidDeployConfig)
+	}
+	return nil
+}
+
```

Comment:
This is missing a godoc

Reply:
Added a godoc to the struct. The `Check` function implements `ConfigChecker`, which is asserted right above, and the same for all config structs. I'm not going to copy-paste go docs, and instead prefer documenting the interface itself.